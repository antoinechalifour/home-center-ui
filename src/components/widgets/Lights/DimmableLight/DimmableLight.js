import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import InputRange from 'ui/InputRange'
import SwitchableLight from '../SwitchableLight'

export default function DimmableLight (props) {
  return (
    <div>
      <SwitchableLight {...props} />
      <InputWrapper>
        <InputRange value={props.bri} onChange={props.updateBrightness} />
      </InputWrapper>
    </div>
  )
}

DimmableLight.propTypes = {
  bri: PropTypes.number.isRequired,
  updateBrightness: PropTypes.func.isRequired
}

const InputWrapper = styled.div`
  padding-top: 12px;
  width: 100%;
  margin: auto;
`
