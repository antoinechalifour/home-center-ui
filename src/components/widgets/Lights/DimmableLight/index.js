import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { graphql, compose } from 'react-apollo'
import debounce from 'debounce'
import { setBrightness, updateLightInformation } from 'queries/lights'
import DimmableLight from './DimmableLight'

export class DimmableLightContainer extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    bri: PropTypes.number.isRequired,
    setBrightness: PropTypes.func.isRequired
  }

  constructor (props) {
    super(props)

    this.state = {
      value: this.props.bri
    }

    this._commitBrightness = debounce(this._commitBrightness.bind(this), 500)
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.bri) {
      this.setState({ value: nextProps.bri })
    }
  }

  _onUpdateBrightness = value => {
    this.setState({ value }, this._commitBrightness)
  }

  _commitBrightness = () => {
    this.props.setBrightness({
      variables: {
        input: {
          id: this.props.id,
          bri: Math.round(this.state.value)
        }
      }
    })
  }

  render () {
    return (
      <DimmableLight
        {...this.props}
        bri={this.state.value}
        updateBrightness={this._onUpdateBrightness}
      />
    )
  }
}

const setBrightnessOptions = {
  name: 'setBrightness'
}

const updateLightInformationOptions = {
  name: 'updateLightInformation',
  options: {
    refetchQueries: ['GetLights']
  }
}

export default compose(
  graphql(setBrightness, setBrightnessOptions),
  graphql(updateLightInformation, updateLightInformationOptions)
)(DimmableLightContainer)
