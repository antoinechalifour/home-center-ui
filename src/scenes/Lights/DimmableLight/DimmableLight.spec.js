import React from 'react'
import { shallow } from 'enzyme'
import DimmableLight from './DimmableLight'

describe('scnees/HomeControl/Lights/DimmableLight', () => {
  it('Should render correctly', () => {
    const props = {
      bri: 33,
      updateBrightness: jest.fn()
    }
    const wrapper = shallow(<DimmableLight {...props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
