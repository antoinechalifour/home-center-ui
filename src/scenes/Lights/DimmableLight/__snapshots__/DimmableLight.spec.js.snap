// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scnees/HomeControl/Lights/DimmableLight Should render correctly 1`] = `
<div>
  <Apollo(Apollo(SwitchableLightContainer))
    bri={33}
    updateBrightness={[Function]}
  />
  <styled.div>
    <InputRange
      onChange={[Function]}
      value={33}
    />
  </styled.div>
</div>
`;
